### ISSUE TITLE
Google Spreadsheets: when the spreadsheet not found show the service account email address

### ISSUE THREAD

**Issue by arikfr** [getredash/redash#2940]
When trying to query a Google Spreadsheet and it wasn't found, as part of the message to the user show the service account email address.

Reasoning: many times the user tries to share it with their own email address or just not sure what address to share it with.

**Comment by adityashrm21** [comment]
Can you tell a bit more about how to extract the service account email address through jsonkeyfile?

**Comment by arikfr** [comment]
Sure!

https://github.com/getredash/redash/blob/be377b5f594ba1715c429f0182e3ea2e00e22a9b/redash/query_runner/google_spreadsheets.py#L182

^ this is how the key file contents file are being loaded. Then there is a key which is the service account email (I think it's `client_email`).

**Comment by YashJipkate** [comment]
Hi @arikfr, Is this issue still up for grabs? I'd like to work on this.
Thanks!

**Comment by arikfr** [comment]
@YashJipkate thank you, but there are actually two open PRs for this issue. Both need some small changes, but probably there is no need for another implementation :grimacing: 

Would you like to work on something else?

**Comment by kurianbenoy** [comment]
Is this issue still not solved @arikfr ?

**Comment by Dyex719** [comment]
If this issue is still open, can I work on it?

**Comment by arikfr** [comment]
@Dyex719 yes, thanks! But check the old pull requests. At this point I don't remember what were the small changes needed, but you can probably get some ideas there.

**Comment by robokage** [comment]
Hi @arikfr if this issue is still open,  I would like to work on it as part of hacktoberfest 2023.  So please let me know

**Comment by guidopetri** [comment]
@robokage sure, feel free to open a PR and tag me on it :)

**Comment by robokage** [comment]
> @robokage sure, feel free to open a PR and tag me on it :)

Great I'll start working on it

**Comment by mohammed-anwar404** [comment]
hey @arikfr This is my first time participating in Hacktoberfest. I have python as my primary skill and would definitely want to contribute.
Please let me know if this issue can be assigned to me.

**Comment by mohammed-anwar404** [comment]
Hey, I am new to hactoberfest looking to contribute to my first project. If this is a beginner-friendly issue I would like to contribute with some guidance from your side.
